standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1293,19 +1304,41 @@ public AllocateTableWriteIdsResponse allocateTableWriteIds(AllocateTableWriteIds
         lockInternal();
         dbConn = getDbConn(Connection.TRANSACTION_READ_COMMITTED);
         stmt = dbConn.createStatement();
+        long writeId;
+        String s;
+
+        handle = getMutexAPI().acquireLock(MUTEX_KEY.WriteIdAllocator.name());
 
         if (rqst.isSetReplPolicy()) {
-          txnIds = getValidTxnIds(rqst, stmt, rs);
+          assert (rqst.isSetSrcTxnToWriteIdList());
+          assert (!rqst.isSetTxnIds());
+          txnIds = checkAndGetTargetTxnIds(rqst, stmt, rs);
+          writeId = rqst.getSrcTxnToWriteIdList().get(0).getWriteId();
         } else {
+          assert (!rqst.isSetSrcTxnToWriteIdList());
```

Comment:
Expect no change in normal allocate write id flow. 
Lets follow the logic of
if(replPolicy) {
txnIds = getTargettxnIds()
if (txnIds.size != srctxnIds.size) return.
}
Allocate write id normal flow.
if (replPolicy) {
validateWriteIdsAgainstSrc()
}

Reply:
done